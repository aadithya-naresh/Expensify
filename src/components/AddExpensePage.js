import React from 'react'

const AddExpensePage = () => (
    <div>
        This is from the add page
    </div>
)

export default AddExpensePage
