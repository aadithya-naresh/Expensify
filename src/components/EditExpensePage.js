import React from 'react'

const EditExpensepage = (props) =>{
    console.log(props)
    return(
    <div>
       Editing {props.match.params.id}
    </div>
    )
}
export default EditExpensepage