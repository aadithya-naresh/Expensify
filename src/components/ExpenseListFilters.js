import React from 'react';
import { connect } from 'react-redux';
import { setTextFilter } from '../actions/filters';

const ExpenseListFilters = (props) => (
  <div>
    <input type="text" value={props.filters.text} onChange={(e) => {
      props.dispatch(setTextFilter(e.target.value))
      console.log(props.state.filters)
    }} />
  </div>
);

const mapStateToProps = (state) => {
  return {
    filters: state.filters,
    state
  };
};

export default connect(mapStateToProps)(ExpenseListFilters);
