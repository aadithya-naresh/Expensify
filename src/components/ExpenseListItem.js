import React from 'react';
import {Link} from 'react-router-dom'
import moment from 'moment'

export const ExpenseListItem = ({ id, description, amount, createdAt }) => (
    <Link to={`/edit/${id}`} className="list-item">
      <div>
        <h3 className="list-item__title">{description}</h3>
        <span className="list-item__sub-title"> {moment(createdAt).format('MMMM Do, YYYY')}</span>
      </div>
        <h3 className="list-item__data"> {'â‚¹'+amount.toLocaleString('en-IN')} </h3>
    </Link>
);

export default ExpenseListItem;