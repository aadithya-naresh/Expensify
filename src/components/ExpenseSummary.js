import React from 'react'
import { connect } from 'react-redux'
import selectExpenses from '../selectors/expenses'
import expensesTotal from '../selectors/expenses-total'

export const ExpenseSummary = (props) =>{
    const expenseWord = props.expensesCount == 1 ? 'expense' : 'expenses'
   
   return( 
    <div>
     <h1>{<p> Viewing {props.expensesCount} {expenseWord} totalling {'â‚¹'+props.expensesTotal.toLocaleString('en-IN')}</p>} </h1>
    </div>
    )
}

const mapStateToProps = (state) => {
    return {
      expensesCount:selectExpenses(state.expenses, state.filters).length,
      expensesTotal: expensesTotal(state.expenses)
    }
}
export default connect(mapStateToProps)(ExpenseSummary);