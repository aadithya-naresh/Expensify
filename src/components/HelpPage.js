import React from 'react'

const HelpPage = () =>(
    <div>
        This is from help page
    </div>
)  

export default HelpPage