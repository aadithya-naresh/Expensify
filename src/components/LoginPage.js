import React from 'react'
import { connect } from 'react-redux'
import {startLogin} from '../actions/auth'

const LoginPage = ({ startLogin }) =>(
    <div>
        <h1>Login Page</h1>
        <button onClick={startLogin}>Login Button</button>
    </div>
)
const mapDispatchToProps = (dispatch) =>({
    startLogin : () => dispatch(startLogin())
})
export default connect(undefined,mapDispatchToProps)(LoginPage)