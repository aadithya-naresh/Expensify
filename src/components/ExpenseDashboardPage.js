import React from 'react'
import ExpenseList from'./ExpenseList'
import ExpenseListFilters from './ExpenseListFilters'
import ExpenseSummary from './ExpenseSummary'

export const ExpenseDashboardPage = () => (
    <div>
    <ExpenseSummary />
    <ExpenseListFilters />
       <ExpenseList />
    </div>
)

export default ExpenseDashboardPage