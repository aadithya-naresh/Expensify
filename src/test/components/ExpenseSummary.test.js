import React from 'react'
import {ExpenseSummary} from '../../components/ExpenseSummary'
import {shallow} from 'enzyme'
import expenses from '../fixtures/expenses'

test('Should render Expense Summary in dashboard', () =>{
    const wrapper = shallow(<ExpenseSummary expensesCount={5} expensesTotal={150} />)
    expect(wrapper).toMatchSnapshot()
})

test('Should render Expense Summary in dashboard', () =>{
    const wrapper = shallow(<ExpenseSummary expensesCount={5} expensesTotal={150} />)
    expect(wrapper).toMatchSnapshot()
})