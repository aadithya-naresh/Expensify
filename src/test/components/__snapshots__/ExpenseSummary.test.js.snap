// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Expense Summary in dashboard 1`] = `
<div>
  <h1>
    <p>
       Viewing 
      5
       
      expenses
       totalling 
      ₹150
    </p>
     
  </h1>
</div>
`;

exports[`Should render Expense Summary in dashboard 2`] = `
<div>
  <h1>
    <p>
       Viewing 
      5
       
      expenses
       totalling 
      ₹150
    </p>
     
  </h1>
</div>
`;
